pkgs/development/tools/misc/segger-jlink/default.nix

```diff
@@ -0,0 +1,160 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, udev
+, config
+, acceptLicense ? config.segger-jlink.acceptLicense or false
+, fontconfig
+, xorg
+}:
+
+let
+  supported = {
+    x86_64-linux = {
+      name = "x86_64";
+      sha256 = "cb43e8656cc2d7c7b42a5f978d065614378e0ef388ef70321ef18c1082450e6a";
+    };
+    i686-linux = {
+      name = "i386";
+      sha256 = "7969354639d8251a7b341bb281649836778bb2ed61166a23779c8db7edc62f54";
+    };
+    aarch64-linux = {
+      name = "arm64";
+      sha256 = "9c1126fae4678df4d6a8e3ee5a1784366efa12050906c49fa95906eb0578ef1d";
+    };
+    armv7l-linux = {
+      name = "arm";
+      sha256 = "e2608cd2f1b666f9e93609e3aa3f6559c7d6f3d9097e84d439c982b23018b1a1";
+    };
+  };
+
+  platform = supported.${stdenv.system} or (throw "unsupported platform ${stdenv.system}");
+
+  version = "792e";
+
+  url = "https://www.segger.com/downloads/jlink/JLink_Linux_V${version}_${platform.name}.tgz";
+
+  src =
+    assert !acceptLicense -> throw ''
+      Use of the "SEGGER JLink Software and Documentation pack" requires the
+      acceptance of the following licenses:
+
+        - SEGGER Downloads Terms of Use [1]
+        - SEGGER Software Licensing [2]
+
+      You can express acceptance by setting acceptLicense to true in your
+      configuration. Note that this is not a free license so it requires allowing
+      unfree licenses as well.
+
+      configuration.nix:
+        nixpkgs.config.allowUnfree = true;
+        nixpkgs.config.segger-jlink.acceptLicense = true;
+
+      config.nix:
+        allowUnfree = true;
+        segger-jlink.acceptLicense = true;
+
+      [1]: ${url}
+      [2]: https://www.segger.com/purchase/licensing/
+    '';
+      fetchurl {
+        inherit url;
+        inherit (platform) sha256;
+        curlOpts = "--data accept_license_agreement=accepted";
+      };
+
+  qt4-bundled = stdenv.mkDerivation {
```

Comment:
Why have this as a second derivation and not just as part of the main one?

Reply:
I wanted to ensure that QT4 libs (which are considered outdated) are not contained in /bin or /lib of a top-level package, in order to not pollute the global namespace. Am I mistaken? Should this be included?

Reply:
I think that makes sense